ecl2df/common.py

```diff
@@ -651,43 +656,107 @@ def stack_on_colnames(
     return dframe
 
 
-def parse_zonemapfile(filename: str) -> Optional[dict]:
-    """Return a dictionary from (int) K layers in the simgrid to strings
+def is_color(input_string: str) -> bool:
+    """Checks if the input string is avalid color.
+    That is six-digit hexadecimal, three-digit hexadecimal or
+    given as a SVG color keyword name
+    """
+    if input_string.lower() in SVG_COLOR_NAMES:
+        return True
+
+    regex = "^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$"
+    return bool(re.match(regex, input_string))
+
+
+def parse_lyrfile(filename: str) -> Optional[List[Dict[str, Any]]]:
+    """Return a list of dicts representation of the lyr file.
 
-    Typical usage is to map from grid layer to zone names.
+    The lyr file contains data of the following format,
+    where the color code is optional::
 
-    The layer filename must currently follow format:
+      'ZoneA' 1-4     #FFE5F7
+      'ZoneB' 5       red
 
-      'ZoneA' 1-4
-      'ZoneB' 5-10
+    Description of the lyr format:
+    https://resinsight.org/3d-main-window/formations/#formation-names-description-files-_lyr_
 
-    where the single quotes are optional for zones without spaces.
-    Write single layer zones as 11-11. NB: ResInsight requires single
-    quotes always.
+    The output has the following format::
 
-    Either "--" or "#" can be used to denote comments.
+        [
+            {
+                "name": "ZoneA",
+                "from_layer": 1,
+                "to_layer": 4,
+                "color": "#FFE5F7"
+            },
+            {
+                "name": "ZoneB",
+                "span": 5,
+                "color": "red"
+            }
+        ]
 
     Args:
-        filename (str): Absolute path to a zone map file (lyr format)
+        filename (str): Absolute path to a lyr file
```

Comment:
`filename (str): Absolute` -> `filename: Absolute` 

sphinx autotyping takes care of stating that the type is a striing.